includes/wikia/services/UserService.class.php

```diff
@@ -21,6 +21,29 @@ public static function getNameFromUrl($url) {
 	}
 
 	/**
+	 * get main page for current user respecting user preferences
+	 * @return Title
+	 */
+	public static function getMainPage() {
+		global $wgUser;
+
+		$title = Title::newMainPage();
+
+		if( $wgUser->isLoggedIn() ) {
+			$value = $wgUser->getGlobalPreference(UserPreferencesV2::LANDING_PAGE_PROP_NAME);
+			switch($value) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a "case default" clause to this "switch" statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement with "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1301)